### ISSUE TITLE
Plugin Scanner can fail if "Permission denied" when trying to access files

### ISSUE THREAD

**Issue by indyrikki** [fcpxhacks/fcpxhacks#2525]
Recently updated Command Post and I'm not sure whether that's the reason, but it's taken to hanging in Scan mode. 
I'd welcome any suggestions as to what I could try in order to fix this, as I'm not a developer, just a very grateful user. 
Here's the debug info which might help?
Build Date:                     Dec 24 2020, 17:56:12
macOS Version:                  10.14.6
CommandPost Locale:             en_GB
Current Keyboard Layout:        British
Final Cut Pro Path:             /Applications/Final Cut Pro.app
Final Cut Pro Version:          10.4.10
Final Cut Pro Language:         en
Developer Mode:                 false

2020-12-29 22:56:45: 2020-12-29 22:56:43: -- Loading /Applications/CommandPost.app/Contents/Resources/extensions/cp/init.lua
2020-12-29 22:56:45: 22:56:45         cp:     Loading Plugins...
2020-12-29 22:56:50: 22:56:50 ** Warning:   plugins: Slow Plugin: core.midi.manager (1.7118802070618)
2020-12-29 22:56:50: 22:56:50 ** Warning:   plugins: Slow Plugin: finalcutpro.midi.controls.controlui.absolute (1.7172529697418)
2020-12-29 22:56:53: 22:56:53 ** Warning:   plugins: Slow Plugin (Post): finalcutpro.touchbar.manager (1.5381698608398)
2020-12-29 22:56:54: 22:56:54         cp:     Plugins Loaded.
2020-12-29 22:56:54:                          Startup Time: 8.353457 seconds
2020-12-29 22:56:54: -- Done.
2020-12-29 22:57:05: 22:57:05 ERROR:   LuaSkin: hs.timer callback error: ...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:713: cannot open /Library/Application Support/Final Cut Pro/Templates.localized/Generators.localized/CX Corporate/Applicable Elegance: Permission denied
stack traceback:
	[C]: in function 'hs.fs.internal.dir'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:713: in function 'cp.apple.finalcutpro.plugins._getPluginName'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:806: in method 'handlePluginDirectory'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:889: in method 'scanPluginCategoryDirectory'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:858: in method 'scanPluginTypeDirectory'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:786: in method 'scanPluginsDirectory'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:1211: in method 'scanSystemMotionTemplates'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:1602: in method 'scanSystemPlugins'
	...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:1663: in function <...ts/Resources/extensions/cp/apple/finalcutpro/plugins.lua:1646>
	(...tail calls...)
	...pp/Contents/Resources/plugins/finalcutpro/setup/scan.lua:55: in function <...pp/Contents/Resources/plugins/finalcutpro/setup/scan.lua:54>

**Comment by latenitefilms** [comment]
For some reason it looks like CommandPost doesn't have permission to access:

```
/Library/Application Support/Final Cut Pro/Templates.localized/Generators.localized/CX Corporate/Applicable Elegance
```

Maybe try changing the permissions for the Templates folder and it's contents to be readable, or try moving these templates into your user folder? Do could also try repairing permissions via Disk Utility, but not sure it'll help.

**Comment by latenitefilms** [comment]
Related: https://github.com/Hammerspoon/hammerspoon/issues/2638

**Comment by indyrikki** [comment]
Before I saw your response I tried the following: 
I copied /Application Support/CommandPost folder somewhere safe, along with the com.latenightfilms preference file.
Then I used CommandPost uninstall.
Next I used a TimeMachine backup to restore my entire Motion Templates folder.
Interestingly this last process seemed to hang too and while it was hanging, my Movies folder showed a greyed out 'Templates.localized' file. I quit the process after a good few minutes and everything appeared as per normal in the Movies folder ie the Motion Templates folder reappeared with all its subfolders seemingly intact. 
I sparked up FCPX and got a warning to restart in order to access all the plugins used in the project. Did that, and everything seemed ok then.
Next, I moved the saved CommandPost folder and file to their respective positions and re-installed CommandPost.
This time on launch, it didn't ask to rescan, and I used my shortcut to launch FCPX - it worked fine. Then tested my shortcut for HUD and everything appeared as per normal along with some plugins I've set up on the Touch Bar.

After seeing your response, I did a bit more detective work. The Applicable Elegance file WAS set to system: 'Read&Write' but Everyone: 'No Access'. 

I tried to change permissions using the Info box, and I found that it hung while "Applying Privileges to 'Applicable Elegance'".

Relaunching Finder cured the hang, and the file's Everyone privilege appeared to be fixed to 'Read Only'.

Launching CommandPost I was asked to scan again, which went without issue, and now everything seems to be working fine.

Thanks for your help, and I hope my relaying my own experience might help someone else. If you have any thoughts on why or how this particular file got to behave so weirdly including the hanging privileges change, then I'd be interested, but otherwise, fingers crossed, case closed.
